md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ProcessAPI.java

```diff
@@ -897,25 +972,73 @@ public Object execute(String[] params) {
     }
 
 
-
     @RequestMapping(value = {"/SLAMonitoring/{id}", "/SLAMonitoring/{id}/"}, method = RequestMethod.GET)
-    @ResponseBody public
-    RestWrapper SLAMonitoring(@PathVariable("id") Integer processId
+    @ResponseBody
+    public RestWrapper SLAMonitoring(@PathVariable("id") Integer processId
     ) {
         RestWrapper restWrapper = null;
-       try {
-           LOGGER.info("processid for slaMonitoring is " + processId);
-           List<com.wipro.ats.bdre.md.dao.jpa.Process> subProcessList = processDAO.subProcesslist(processId);
-           LOGGER.info("number of subprocesses is " + subProcessList.size());
-           List<SLAMonitoringBean> slaMonitoringBeanList = instanceExecDAO.slaMonitoringData(subProcessList);
-           LOGGER.info("total size of slaMonitoringBeanList " + slaMonitoringBeanList.size());
-           LOGGER.info("slaMonitoringBean are "+slaMonitoringBeanList);
-
-           restWrapper = new RestWrapper(slaMonitoringBeanList, RestWrapper.OK);
-       }catch (MetadataException e) {
-           LOGGER.error(e);
-           restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
-       }
-         return restWrapper;
+        try {
+            LOGGER.info("processid for slaMonitoring is " + processId);
+            List<com.wipro.ats.bdre.md.dao.jpa.Process> subProcessList = processDAO.subProcesslist(processId);
+            LOGGER.info("number of subprocesses is " + subProcessList.size());
+            List<SLAMonitoringBean> slaMonitoringBeanList = instanceExecDAO.slaMonitoringData(subProcessList);
+            LOGGER.info("total size of slaMonitoringBeanList " + slaMonitoringBeanList.size());
+            LOGGER.info("slaMonitoringBean are " + slaMonitoringBeanList);
+
+            restWrapper = new RestWrapper(slaMonitoringBeanList, RestWrapper.OK);
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+
+    @RequestMapping(value = {"/options", "/options/"}, method = RequestMethod.POST)
+    @ResponseBody
+    public RestWrapperOptions listOptions(Principal principal) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused method parameter(s) "principal". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)